drivers/sensor/st/lsm9ds1/lsm9ds1.c

```diff
@@ -0,0 +1,651 @@
+/*
+ * Copyright (c) 2024 Bootlin
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_lsm9ds1
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+
+#include "lsm9ds1.h"
+#include <stmemsc.h>
+
+LOG_MODULE_REGISTER(LSM9DS1, CONFIG_SENSOR_LOG_LEVEL);
+
+/* Sensitivity of the accelerometer, indexed by the raw full scal value. Unit is µg/ LSB */
+static const uint16_t lsm9ds1_accel_fs_sens[] = {61, 732, 122, 244};
+
+/*
+ * Sensitivity of the gyroscope, indexed by the raw full scale value.
+ * The value here is just a factor applied to GAIN_UNIT_G, as the sensitivity is
+ * proportional to the full scale size.
+ * The index 2 is never used : the value `0` is just a placeholder.
+ */
+static const uint16_t lsm9ds1_gyro_fs_sens[] = {1, 2, 0, 8};
+
+/*
+ * Values of the different sampling frequencies of the accelerometer, indexed by the raw odr
+ * value that the sensor understand.
+ */
+static const uint16_t lsm9ds1_odr_map[] = {0, 10, 50, 119, 238, 476, 952};
+
+/*
+ * Value of the different sampling frequencies of the gyroscope, indexed by the raw odr value
+ * that the sensor understand.
+ */
+static const uint16_t lsm9ds1_gyro_odr_map[] = {0, 15, 59, 119, 238, 476, 952};
+
+static int lsm9ds1_reboot(const struct device *dev)
+{
+	const struct lsm9ds1_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lsm9ds1_ctrl_reg8_t ctrl8_reg;
+	int ret;
+
+	ret = lsm9ds1_read_reg(ctx, LSM9DS1_CTRL_REG8, (uint8_t *)&ctrl8_reg, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ctrl8_reg.boot = 1;
+
+	ret = lsm9ds1_write_reg(ctx, LSM9DS1_CTRL_REG8, (uint8_t *)&ctrl8_reg, 1);
+	if (ret < 0) {
+		return ret;
+	}
```

Comment:
ok, but why not using the lsm9ds1_dev_boot_set()?
I'm not against your approach anyway

Reply:
yes, agree